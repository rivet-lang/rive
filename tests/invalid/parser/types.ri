extern (C) {
    fn malloc(sz: usize) *void;
    fn free(pointer: &void); // BAD
}

struct St {
    f1: &&i32;
    f2: *&i32;
    f3: &*i32; // OK

    f4: ?*u8;

    f5: (i32, i32, i32, i32, i32, i32, i32, i32, i32);

    f6: ??i32;

    f7: void;

    f8: pkg;
    f9: =;
}

fn main() { }
