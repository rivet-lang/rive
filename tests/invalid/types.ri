extern (C) {
    fn malloc(sz: usize) *void;
    fn free(pointer: &void) void; // FAIL
}

struct St {
    f1: &i32; // FAIL

    f2: (i32, i32, i32, i32, i32, i32, i32, i32, i32);

    f3: void;

    f4: pkg;
    f5: =;
}

fn main() { }
