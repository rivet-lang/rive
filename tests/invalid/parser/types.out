tests/failing/parser/types.ri:3:22: error: cannot use references inside `extern` blocks
    |
  3 |     fn free(pointer: &void); // BAD
    |                      ^
    = help: use pointers instead: `*const void`
tests/failing/parser/types.ri:7:9: error: multi-level references are not allowed
    |
  7 |     f1: &&i32;
    |         ^
tests/failing/parser/types.ri:8:9: error: cannot use pointers with references
    |
  8 |     f2: *&i32;
    |         ^
tests/failing/parser/types.ri:11:9: error: pointers cannot be optional
     |
  11 |     f4: ?*u8;
     |         ^
     = note: by default pointers can contain the value `none`
tests/failing/parser/types.ri:13:9: error: tuples can have a maximum of 8 types
     |
  13 |     f5: (i32, i32, i32, i32, i32, i32, i32, i32, i32);
     |         ^
     = help: you can use a struct instead
tests/failing/parser/types.ri:15:9: error: optional multi-level types are not allowed
     |
  15 |     f6: ??i32;
     |         ^
tests/failing/parser/types.ri:17:9: error: invalid use of `void` type
     |
  17 |     f7: void;
     |         ^
tests/failing/parser/types.ri:19:9: error: expected type, found keyword `pkg`
     |
  19 |     f8: pkg;
     |         ^
tests/failing/parser/types.ri:20:9: error: expected type, found token `=`
     |
  20 |     f9: =;
     |         ^
rivetc: error: could not compile package `core`, aborting due to 9 previous errors